content/docs/2_cookbook/9_setup/0_kirby-meets-purgecss/cookbook-recipe.txt

```diff
@@ -0,0 +1,151 @@
+.htaccess
+purge/
+```
+
+## Node.js Configuration
+Also in the projects' root, we create a `package.json` file that controls the build process. This file contains only a (*build*) script.
+
+```js "/package.json"
+{
+  "name": "projectname",
+	"scripts": {
+    "build": "npx purgecss -css assets/css/CSS_FRAMEWORK.css --content site/**/*.php --output purge/"
+	}
+}
+```
+
+### Explanation
+Use `build` to generate a final minified CSS file
+
+- `--css assets/css/CSS_FRAMEWORK.css`: input file with your framework's CSS
+- `--content 'site/**/*.php'`: folder to watch for CSS framework's classes
+- `--output  purge/`: output folder with purged CSS
+
+If you want to customize your PurgeCSS installation, you can create an additional config file `purgecss.config.js` in the project root. This file will then be automatically processed at build time if it exists.
+
+```js "/purgecss.config.js"
+module.exports = {
+    css: ['assets/css/CSS_FRAMEWORK.css'],
+    content: ['site/**/*.php'],
+    output: 'purge/'
+}
+```
+
+<info>
+More details on how to customize PurgeCSS can be found (link: https://purgecss.com/configuration.html text: in the PurgeCSS docs).
+</info>
+
+## Build
+Now we have a fully functional basic setup and are ready to generate our purged CSS file. Make sure you are in the project root folder and start the `build` script as follows:
+
+```bash
+npm run build
+```
+
+## Setup with GUI
+
+Before we start, please make sure Prepros is installed on your local machine. You can download the free unlimited trial from (link: https://prepros.io/downloads text: the official Prepros.io website). Prepros compiles your files, transpiles your JavaScript and provides a graphical user interface like (link: https://codekitapp.com/ text: CodeKit) for Mac or (link: https://scout-app.io/ text: Scout) (only SASS & SCSS).
+
+
+## Prepros Configuration
+
+<warning>
+Make sure you add the root folder of your Kirby installation to Prepros as a new project, e.g. `starterkit/` otherwise the custom commands in Prepros will not work as intended since they refer to relative paths! 
+</warning>
+
+<info>
+Prepros will likely popup an error regarding that there are more than 2000 files for the watcher. Just close the popup since there **is no need** to increase the limit. Afterward, Prepros will automatically disable "*Watch files for changes*". Simply toggle the setup at least for your CSS folders.
+</info>
+
+1. Open _Project Settings_ 
+2. Navigate to _CSS Tools_
+3. Select _CSS_ or _Sass_ 
+4. Scroll to _Tasks_ 
+5. Type the following command:
+
+```bash
+purgecss --css {{input}} --content "{{rawRootDir}}/site/**/*.php" --output {{outputDir}} {{output}}
+```
+
+6. Click on _Add Custom Command_
+
+(image: prepros-config-screenshot.png caption: Screenshot of Prepros "Add Custom Command" section)
+
+<info>
+`{{output}}` placeholder is required by Prepros but PurgeCSS doesn't support explicit output path so the `{{output}}` placeholder is ignored by PurgeCSS.
```

Comment:
The `{{output}}` placeholder is required by Prepros but PurgeCSS doesn't support an explicit output path, so the `{{output}}` placeholder is ignored by PurgeCSS.